adapters/relevantdigital/relevantdigital.go

```diff
@@ -0,0 +1,278 @@
+package relevantdigital
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"strings"
+	"text/template"
+
+	"github.com/buger/jsonparser"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+	name     string
+}
+
+const RELEVANT_DOMAIN = ".relevant-digital.com"
+const DEFAULT_TIMEOUT = 1000
+const DEFAULT_BUFFER_MS = 250
+const STORED_REQUEST_EXT = "{\"prebid\":{\"debug\":%t,\"storedrequest\":{\"id\":\"%s\"}},\"relevant\":{\"count\":%d,\"adapterType\":\"server\"}}"
+const STORED_IMP_EXT = "{\"prebid\":{\"storedrequest\":{\"id\":\"%s\"}}}"
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	return &adapter{
+		endpoint: template,
+		name:     bidderName.String(),
+	}, nil
+}
+
+func patchBidRequestExt(prebidBidRequest *openrtb2.BidRequest, id string) error {
+	count, cerr := jsonparser.GetInt(prebidBidRequest.Ext, "relevant", "count")
+	if cerr != nil {
+		count = 0
+	}
+
+	if count >= 5 {
+		return &errortypes.FailedToRequestBids{
+			Message: "too many requests",
+		}
+	} else {
+		count = count + 1
+	}
+
+	debug, derr := jsonparser.GetBoolean(prebidBidRequest.Ext, "prebid", "debug")
+	if derr != nil {
+		debug = false
+	}
+
+	prebidBidRequest.Ext = []byte(fmt.Sprintf(STORED_REQUEST_EXT, debug, id, count))
+	return nil
+}
+
+func patchBidImpExt(imp *openrtb2.Imp, id string) {
+	imp.Ext = []byte(fmt.Sprintf(STORED_IMP_EXT, id))
+	if imp.Banner != nil {
+		imp.Banner.Ext = nil
+	}
+	if imp.Video != nil {
+		imp.Video.Ext = nil
+	}
+	if imp.Native != nil {
+		imp.Native.Ext = nil
+	}
+	if imp.Audio != nil {
+		imp.Audio.Ext = nil
```

Comment:
any specifc reasion to set `imp.[Banner/ Video/  Native / Audio].Ext` to nil?

Reply:
Fixed